test/account/PerHookData.t.sol

```diff
@@ -3,63 +3,28 @@ pragma solidity ^0.8.25;
 
 import {PackedUserOperation} from "@eth-infinitism/account-abstraction/interfaces/PackedUserOperation.sol";
 import {IEntryPoint} from "@eth-infinitism/account-abstraction/interfaces/IEntryPoint.sol";
-import {ERC1967Proxy} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
 import {MessageHashUtils} from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
 
 import {UpgradeableModularAccount} from "../../src/account/UpgradeableModularAccount.sol";
-import {ISingleOwnerPlugin} from "../../src/plugins/owner/ISingleOwnerPlugin.sol";
 import {FunctionReference, FunctionReferenceLib} from "../../src/helpers/FunctionReferenceLib.sol";
 
 import {MockAccessControlHookPlugin} from "../mocks/plugins/MockAccessControlHookPlugin.sol";
 import {Counter} from "../mocks/Counter.sol";
-import {AccountTestBase} from "../utils/AccountTestBase.sol";
+import {CustomValidationTestBase} from "../utils/CustomValidationTestBase.sol";
 
-contract PerHookDataTest is AccountTestBase {
+contract PerHookDataTest is CustomValidationTestBase {
     using MessageHashUtils for bytes32;
 
     MockAccessControlHookPlugin internal accessControlHookPlugin;
 
     Counter internal counter;
 
-    FunctionReference internal ownerValidation;
-
-    uint256 public constant CALL_GAS_LIMIT = 50000;
-    uint256 public constant VERIFICATION_GAS_LIMIT = 1200000;
-
     function setUp() public {
         counter = new Counter();
 
         accessControlHookPlugin = new MockAccessControlHookPlugin();
 
-        // Write over `account1` with a new account proxy, with different initialization.
-
-        address accountImplementation = address(factory.accountImplementation());
-
-        account1 = UpgradeableModularAccount(payable(new ERC1967Proxy(accountImplementation, "")));
-
-        ownerValidation = FunctionReferenceLib.pack(
-            address(singleOwnerPlugin), uint8(ISingleOwnerPlugin.FunctionId.VALIDATION_OWNER)
-        );
-
-        FunctionReference accessControlHook = FunctionReferenceLib.pack(
-            address(accessControlHookPlugin), uint8(MockAccessControlHookPlugin.FunctionId.PRE_VALIDATION_HOOK)
-        );
-
-        FunctionReference[] memory preValidationHooks = new FunctionReference[](1);
-        preValidationHooks[0] = accessControlHook;
-
-        bytes[] memory preValidationHookData = new bytes[](1);
-        // Access control is restricted to only the counter
-        preValidationHookData[0] = abi.encode(counter);
-
-        bytes memory packedPreValidationHooks = abi.encode(preValidationHooks, preValidationHookData);
-
-        vm.prank(address(entryPoint));
-        account1.installValidation(
-            ownerValidation, true, new bytes4[](0), abi.encode(owner1), packedPreValidationHooks
-        );
-
-        vm.deal(address(account1), 100 ether);
+        _customValidationSetup();
```

Comment:
Do the old tests here test allowList fails?
Are the error thrown from `AllowlistPlugin` captured as AA errors?

Reply:
Yes, depending on the call path, they throw either user op or runtime errors, and the fuzzer expects them to be called.

The error to look for is loaded from here: https://github.com/erc6900/reference-implementation/blob/8c7133c2b5ad3239a385deb840750dc66fd05e1d/test/samples/AllowlistPlugin.t.sol#L142-L199

And the test expects them here: https://github.com/erc6900/reference-implementation/blob/8c7133c2b5ad3239a385deb840750dc66fd05e1d/test/utils/AccountTestBase.sol#L168 https://github.com/erc6900/reference-implementation/blob/8c7133c2b5ad3239a385deb840750dc66fd05e1d/test/utils/AccountTestBase.sol#L113